Services/source/housekeeping/housekeeping_service.c

```diff
@@ -1,975 +1,993 @@
+
+      if (!csp_send(conn, packet, 50)) {
+        csp_buffer_free(packet);
+      }
+      break;
+    
+    case GET_MAX_FILES:
+      new_max_files = MAX_FILES;
+      new_max_files = csp_hton16(new_max_files);
+      status = 0;
+      memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+      memcpy(&packet->data[OUT_DATA_BYTE], &new_max_files, sizeof(new_max_files));
+
+      set_packet_length(packet, sizeof(int8_t)
+                              + sizeof(new_max_files)
+                              + 1);  // +1 for subservice
+
+      if (!csp_send(conn, packet, 50)) {
+        csp_buffer_free(packet);
+      }
+      break;
+
+    case GET_HK:
+      data16 = (uint16_t *)(packet->data + 1);
+      limit = data16[0];
+      before_id = data16[1];
+      before_time = ((uint32_t *)data16)[1];
+      
+      if (!csp_send(conn, packet, 50)) {
+      csp_buffer_free(packet);
+        if (fetch_historic_hk_and_transmit(conn, limit, before_id, before_time) != SUCCESS) {
+            return SATR_ERROR;
+        }
+      }
+      break;
+
+    default:
+      ex2_log("No such subservice\n");
+      return SATR_PKT_ILLEGAL_SUBSERVICE;
+  }
+
+  return SATR_OK;
+}
+
+SAT_returnState start_housekeeping_service(void);
+
+/**
+ * @brief
+ *      FreeRTOS housekeeping server task
+ * @details
+ *      Accepts incoming housekeeping service packets and executes the application
+ * @param void* param
+ * @return None
+ */
+void housekeeping_service(void *param) {
+    csp_socket_t *sock;
+    sock = csp_socket(CSP_SO_RDPREQ);
+    csp_bind(sock, TC_HOUSEKEEPING_SERVICE);
+    csp_listen(sock, SERVICE_BACKLOG_LEN);
+    svc_wdt_counter++;
+
+    for (;;) {
+        csp_conn_t *conn;
+        csp_packet_t *packet;
+        if ((conn = csp_accept(sock, CSP_MAX_TIMEOUT)) == NULL) {
+          /* timeout */
+          continue;
+        }
+        svc_wdt_counter++;
+
+        while ((packet = csp_read(conn, 50)) != NULL) {
+          if (hk_service_app(conn, packet) != SATR_OK) {
+            ex2_log("Error responding to packet");
+          }
+        }
+        csp_close(conn); // frees buffers used
+    }
+}
+
+/**
+ * @brief
+ *      Start the housekeeping server task
+ * @details
+ *      Starts the FreeRTOS task responsible for accepting incoming
+ *      housekeeping service requests
+ * @param None
+ * @return SAT_returnState
+ *      success report
+ */
+SAT_returnState start_housekeeping_service(void) {
+    TaskHandle_t svc_tsk;
+    taskFunctions svc_funcs = {0};
+    svc_funcs.getCounterFunction = get_svc_wdt_counter;
+    if (xTaskCreate((TaskFunction_t)housekeeping_service, "start_housekeeping_service", 400, NULL,
+                    NORMAL_SERVICE_PRIO, &svc_tsk) != pdPASS) {
+    ex2_log("FAILED TO CREATE TASK start_housekeeping_service\n");
+    return SATR_ERROR;
+  }
+  ex2_register(svc_tsk, svc_funcs);
+  ex2_log("Service handlers started\n");
+  return SATR_OK;
+}
```

Comment:
Looks like this whole file was converted to two space indentation. It should be 4

Reply:
file has been reformatted with clang!